internal/util.go

```diff
@@ -0,0 +1,25 @@
+package internal
+
+import (
+	"reflect"
+)
+
+// RemainingKeys will inspect a struct and compare it to a map. Any key that
+// is defined in a JSON tag of the struct will be removed from the map. The
+// remaining map and keys are then returned.
+//
+// This is useful for determining the extra fields returned in response bodies
+// for resources that can contain an arbitrary or dynamic number of fields.
+func RemainingKeys(s interface{}, m map[string]interface{}) map[string]interface{} {
```

Comment:
At first glance, I'd prefer one of the following:

1. `RemoveKnownKeys(s interface{}, m map[string]interface{})` (this one modifies `m`)
2. `RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]interface{})` (this one doesn't modify `m`)

I'd prefer 2